rpcserver.go

```diff
@@ -970,6 +974,111 @@ func (r *rpcServer) EstimateFee(ctx context.Context,
 	return resp, nil
 }
 
+// unmarshallOutPoint converts an outpoint from its lnrpc type to its canonical
+// type.
+func unmarshallOutPoint(op *lnrpc.OutPoint) (*wire.OutPoint, error) {
+	var hash chainhash.Hash
+	switch {
+	case len(op.TxidBytes) == 0 && len(op.TxidStr) == 0:
+		fallthrough
+
+	case len(op.TxidBytes) != 0 && len(op.TxidStr) != 0:
+		return nil, fmt.Errorf("either TxidBytes or TxidStr must be " +
+			"specified, but not both")
+
+	// The hash was provided as raw bytes.
+	case len(op.TxidBytes) != 0:
+		copy(hash[:], op.TxidBytes)
+
+	// The hash was provided as a hex-encoded string.
+	case len(op.TxidStr) != 0:
+		h, err := chainhash.NewHashFromStr(op.TxidStr)
+		if err != nil {
+			return nil, err
+		}
+		hash = *h
+	}
+
+	return &wire.OutPoint{
+		Hash:  hash,
+		Index: op.OutputIndex,
+	}, nil
+}
+
+// BumpFee allows bumping the fee rate of an arbitrary input. The fee rate can
+// be expressed either as a specific fee rate or a delta of blocks in which the
+// output should be swept on-chain within. If a fee rate is not explicitly
+// specified, then a fee rate corresponding to a confirmation target of 6 blocks
+// will be used. The status of the input sweep can be checked through the
+// PendingSweeps RPC.
+func (r *rpcServer) BumpFee(ctx context.Context,
+	in *lnrpc.BumpFeeRequest) (*lnrpc.BumpFeeResponse, error) {
+
+	// Parse the outpoint from the request.
+	op, err := unmarshallOutPoint(in.Outpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	// Construct the request's fee preference.
+	satPerKw := lnwallet.SatPerKVByte(in.SatPerByte * 1000).FeePerKWeight()
+	feePreference := sweep.FeePreference{
+		ConfTarget: uint32(in.TargetConf),
+		FeeRate:    satPerKw,
+	}
```

Comment:
Validation here in case both or none are set?

Reply:
Validation is done within the `UtxoSweeper.BumpFee` call.

Reply:
Just checking that exactly one of those fields is set could may be done in a `sweeper.NewFeePreference(confTarget, feeRate)` constructor. It simplifies consumers of this struct, because they don't all need to do this check.

Reply:
Some other RPCs rely on a fee preference not being set to choose the default. Adding a constructor that applies additional validation checks would lead to some confusion IMO. Either we should unify the behavior of `FeePreference` all around, or we introduce a new `SweepPreference` type that applies the validation needed for sweeps only.